.changeset/purple-donkeys-camp.md

```diff
@@ -0,0 +1,67 @@
+---
+'@credo-ts/question-answer': minor
+'@credo-ts/bbs-signatures': minor
+'@credo-ts/action-menu': minor
+'@credo-ts/anoncreds': minor
+'@credo-ts/indy-vdr': minor
+'@credo-ts/didcomm': minor
+'@credo-ts/tenants': minor
+'@credo-ts/askar': minor
+'@credo-ts/cheqd': minor
+'@credo-ts/core': minor
+'@credo-ts/drpc': minor
+'@credo-ts/node': minor
+---
+
+DIDComm has been extracted out of the Core. This means that now all DIDComm related modules (e.g. proofs, credentials) must be explicitly added when creating an `Agent` instance. Therefore, their API will be accesable under `agent.modules.[moduleAPI]` instead of `agent.[moduleAPI]`. Some `Agent` DIDComm-related properties and methods where also moved to the API of a new DIDComm module (e.g. `agent.registerInboundTransport` turned into `agent.modules.didcomm.registerInboundTransport`).
```

Comment:
This is not for this PR, but I think before 0.6 we should look at making:
- `agent.didcomm` possible (nesting is getting quite large)
- nest didcomm modules under `agent.didcomm`, also because `agent.credentials` / `agent.modules.credentials` is misleading, since it's didcomm specific. I think registerin
- Adopting the conenctions / oob modules into the didcomm module by default, and you can configure it in the didcomm module config again. I think this simplifies setup.
- Registering custom didcomm protocols on the didcomm module rather than the agent? This may solve some of the initialization and feature registry issues. We can create a `DidcommModule` interface that has special feature registry etc.

Reply:
Yeah good ideas there! Initially I made a "DidCommModule" but I decided to leave it for a further PR to not overcomplicate it. But for sure it is something that makes sense, especially to have a tidy Agent API.

Same thing about the transport config, which is something not so evident for developers.